source/shared/cpp/ObjectModel/MarkDownHtmlGenerator.h

```diff
@@ -215,10 +215,17 @@ class MarkDownRightEmphasisHtmlGenerator : public MarkDownEmphasisHtmlGenerator
         MarkDownEmphasisHtmlGenerator(token, sizeOfEmphasisDelimiterRun, type){};
 
     void GenerateTags(std::shared_ptr<MarkDownEmphasisHtmlGenerator>& token);
+
     bool IsRightEmphasis() const override
     {
-        return true;
+        return m_directionType == Right;
     }
+
+    bool IsLeftEmphasis() const override
```

Comment:
Should these methods from `MarkDownRightEmphasisHtmlGenerator` mirror the methods in `MarkDownLeftEmphasisHtmlGenerator`?

Reply:
I'm not sure; what you mean.

Reply:
The **RightEmphasisHtmlGenerator** class seems very similar to this one:
https://github.com/microsoft/AdaptiveCards/blob/2359e6e5416e78005613e9fc6784b543adca6041/source/shared/cpp/ObjectModel/MarkDownHtmlGenerator.h#L179-L200

So I was curious why we wouldn't just have `IsRightEmphasis() -> return true` here as well.

Reply:
oic; ChangeDirectionToLeft() will change the direction and this change takes account of it;